bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    static {
+        addMapping(CHANNEL_TEMPERATURE, PARAM_TEMPERATURE, CELSIUS);
+        addMapping(CHANNEL_HUMIDITY, PARAM_HUMIDITY, PERCENT);
+        addMapping(CHANNEL_WIND_DIRECTION, PARAM_WIND_DIRECTION, DEGREE_ANGLE);
+        addMapping(CHANNEL_WIND_SPEED, PARAM_WIND_SPEED, METRE_PER_SECOND);
+        addMapping(CHANNEL_GUST, PARAM_WIND_GUST, METRE_PER_SECOND);
+        addMapping(CHANNEL_PRESSURE, PARAM_PRESSURE, MILLIBAR);
+        addMapping(CHANNEL_PRECIPITATION_INTENSITY, PARAM_PRECIPITATION_1H, MILLIMETRE_PER_HOUR);
+        addMapping(CHANNEL_TOTAL_CLOUD_COVER, PARAM_TOTAL_CLOUD_COVER, PERCENT);
+        addMapping(CHANNEL_FORECAST_WEATHER_ID, PARAM_WEATHER_SYMBOL, null);
+    }
+
+    private @NonNullByDefault({}) LatLon location;
+
+    public ForecastWeatherHandler(Thing thing) {
+        super(thing);
+        // Override poll interval to slower value
+        pollIntervalSeconds = (int) TimeUnit.MINUTES.toSeconds(QUERY_RESOLUTION_MINUTES);
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            Object location = getConfig().get(BindingConstants.LOCATION);
+            if (location == null) {
+                throw new IllegalStateException();
+            }
+            String latlon = location.toString();
+            String[] split = latlon.split(",");
+            this.location = null;
+            this.location = new LatLon(new BigDecimal(split[0].trim()), new BigDecimal(split[1].trim()));
+            super.initialize();
+        } catch (IllegalStateException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter not set %s %s", e.getClass().getName(), e.getMessage()));
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter should be in format LATITUDE,LONGITUDE. Error: %s %s",
+                            e.getClass().getName(), e.getMessage()));
+        }
+    }
+
+    @Override
+    protected void update() {
+        if (location == null) {
+            return;
+        }
+        try {
+            long now = Instant.now().getEpochSecond();
+            response = client.query(new ForecastRequest(location, floorToEvenMinutes(now, QUERY_RESOLUTION_MINUTES),
+                    ceilToEvenMinutes(now + TimeUnit.HOURS.toSeconds(FORECAST_HORIZON_HOURS), QUERY_RESOLUTION_MINUTES),
+                    QUERY_RESOLUTION_MINUTES), TIMEOUT_MILLIS);
+        } catch (FMIResponseException e) {
+            // IllegalStateException: Unexpected (possibly bug) issue with response
+            response = null;
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    String.format("{}: {}", e.getClass().getName(), e.getMessage()));
```

Comment:
I'm getting "Status: OFFLINE - COMMUNICATION_ERROR {}: {}" isn't it String.format("%1$s: %2$s", ...)?